fastplotlib/graphics/_positions_base.py

```diff
@@ -293,13 +292,15 @@ def add_rectangle_selector(
 
         ymin = np.floor(y_axis_vals.min()).astype(int)
         ymax = np.ceil(y_axis_vals.max()).astype(int)
+        xmin = np.floor(x_axis_vals.min()).astype(int)
+        xmax = np.ceil(x_axis_vals.max()).astype(int)
 
         # default selection is 25% of the image
         if selection is None:
-            selection = (x_axis_vals[0], x_axis_vals[value_25p], ymin, ymax)
+            selection = (xmin, xmin + 0.25 * (xmax - xmin), ymin, ymax)
 
         # min/max limits
-        limits = (x_axis_vals[0], x_axis_vals[-1], ymin * 1.5, ymax * 1.5)
+        limits = (xmin, xmax, ymin, ymax)
```

Comment:
Why did you remove the `ymin * 1.5` and `ymax * 1.5`? I think we had that there to allow the selector to go a little bit above and below the graphic.

Reply:
Ah sorry - I removed it because with positive ymin the limits didn't allow you to select all the data, restored in 75b7d73c7113c44482b23da9de26d9edddda75ed